### PR TITLE
New GCP fileset to parse cloud armor logs

### PR DESCRIPTION
**Pull request by williamclot**

### CODE REVIEW

File: x-pack/filebeat/module/gcp/cloud_armor/config/pipeline.js [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+        target: "json",
+    });
+
+    // Set @timetamp the LogEntry's timestamp.
+    var parseTimestamp = new processor.Timestamp({
+        field: "json.timestamp",
+        timezone: "UTC",
+        layouts: ["2006-01-02T15:04:05.999999999Z07:00"],
+        tests: ["2019-06-14T03:50:10.845445834Z"],
+        ignore_missing: true,
```
Comment (dan-duffy) [inline x-pack/filebeat/module/gcp/cloud_armor/config/pipeline.js:21]: Do we want to ignore missing in this case given it's a timestamp? Or is this more for testing at the moment?

File: x-pack/filebeat/module/gcp/cloud_armor/config/pipeline.js [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+            {
+                from: "json.httpRequest.remoteIp",
+                to: "gcp.cloud_armor.http_request.remote_ip",
+                type: "ip"
+            },
+            {
+                from: "json.httpRequest.serverIp",
+                to: "gcp.cloud_armor.http_request.server_ip",
+                type: "ip"
+            },
```
Comment (dan-duffy) [inline x-pack/filebeat/module/gcp/cloud_armor/config/pipeline.js:107]: Do you think it's worth having Referrer in this list? So we can attempt to see (where possible) the path taken? Not sure if all that much value from an attack side would be gained so open to leaving this one

**Review by dan-duffy — APPROVED** [review]
Couple of general comments but overall LGTM!

File: x-pack/filebeat/module/gcp/cloud_armor/config/pipeline.js (hunk 1 above)
Comment (williamclot) [reply x-pack/filebeat/module/gcp/cloud_armor/config/pipeline.js:21]: I think I followed what the GCP audit fileset was doing out of the box in the `gcp` module but you're probably right!

File: x-pack/filebeat/module/gcp/cloud_armor/config/pipeline.js (hunk 2 above)
Comment (williamclot) [reply x-pack/filebeat/module/gcp/cloud_armor/config/pipeline.js:107]: Very true, weirdly i didn't have a referer in my log example when writing this! But yes it should be there: https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#HttpRequest

Comment (williamclot) [reply x-pack/filebeat/module/gcp/cloud_armor/config/pipeline.js:107]: Added in https://github.com/unravelin/beats/pull/1/commits/d2333c4d8da374ec528bc59cc3f8f5a087d647bc

**Pull request merged**